zrml/prediction-markets/src/tests.rs

```diff
@@ -3036,16 +3158,21 @@ fn report_fails_if_reporter_is_not_the_oracle() {
             Origin::signed(ALICE),
             BOB,
             MarketPeriod::Timestamp(0..100_000_000),
+            get_deadlines(),
             gen_metadata(2),
             MarketCreation::Permissionless,
             MarketType::Categorical(2),
             MarketDisputeMechanism::SimpleDisputes,
             ScoringRule::CPMM
         ));
+        let market = MarketCommons::market(&0).unwrap();
         set_timestamp_for_on_initialize(100_000_000);
         // Trigger hooks which close the market.
         run_to_block(2);
-        set_timestamp_for_on_initialize(100_056_789);
+        let grace_period: u64 = market.deadlines.grace_period * MILLISECS_PER_BLOCK as u64;
+        set_timestamp_for_on_initialize(
+            100_000_000 + grace_period + MILLISECS_PER_BLOCK as u64 + MILLISECS_PER_BLOCK as u64,
```

Comment:
I am not completely sure but if I remove it I get error, 
```
---- tests::report_fails_if_reporter_is_not_the_oracle stdout ----
thread 'tests::report_fails_if_reporter_is_not_the_oracle' panicked at 'assertion failed: `(left == right)`
  left: `Err(DispatchErrorWithPostInfo { post_info: PostDispatchInfo { actual_weight: None, pays_fee: Pays::Yes }, error: Module(ModuleError { index: 6, error: [28, 0, 0, 0], message: Some("NotAllowedToReportYet") }) })`,
 right: `Err(DispatchErrorWithPostInfo { post_info: PostDispatchInfo { actual_weight: None, pays_fee: Pays::Yes }, error: Module(ModuleError { index: 6, error: [20, 0, 0, 0], message: Some("ReporterNotOracle") }) })`', zrml/prediction-markets/src/tests.rs:3179:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
perhaps its because how function substract MILLISECS_PER_DAY 
```rust
    2 pub fn set_timestamp_for_on_initialize(time: Moment) {                                                         
    1     Timestamp::set_timestamp(time - MILLISECS_PER_BLOCK as u64);                                               
  340 } 
  ```

Reply:
You have to use the normal `Timestamp::set_timestamp` here; `set_timestamp_for_on_initialize` is, well, only when you're reading the timestamp during `on_initialize`. But here, you're operating in a normal extrinsic.

This problem occurs in three places:

- `the_entire_market_lifecycle_works_with_timestamps`
- `full_scalar_market_lifecycle`
- `it_allows_only_oracle_to_report_the_outcome_of_a_market_during_oracle_duration_moment`